Zend/zend_object_handlers.c

```diff
@@ -1502,6 +1532,14 @@ ZEND_API zval *zend_std_get_static_property_with_info(zend_class_entry *ce, zend
 	ret = CE_STATIC_MEMBERS(ce) + property_info->offset;
 	ZVAL_DEINDIRECT(ret);
 
+	if (UNEXPECTED((type == BP_VAR_RW || type == BP_VAR_W) && property_info->flags & ZEND_ACC_FINAL && Z_TYPE_P(ret) != IS_UNDEF)) {
```

Comment:
Hm... I think there's a pretty big problem here :(

We currently compile the operands of property accesses with the property access mode, i.e. `Foo::$bar->baz = 1` is going to perform a W fetch on the static property, rather than the R fetch you would expect.

This means that code like this is going to error, while it should not:

```
<?php
class Foo {
    final public static stdClass $prop;
}

Foo::$prop = new stdClass;
Foo::$prop->bar = 1;
```

Probably more test coverage for modification of objects stored in readonly properties is needed.

Reply:
I tried to fix this in #3865, but ran into some compatibility issues with SimpleXML.

Reply:
Sounds really bad. :/ But I'll add more tests of course. Do you have any idea if this issue is a deal breaker or could be somehow circumvented? Is it worth to have SimpleXML in core if it has issues like this?

Reply:
For the record, I opened #5250 and started a thread at https://externals.io/message/108950.

> Btw is it worth to have SimpleXML in core if it has issues like this?

It's not, but people will complain if we drop it :)